sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/InvocationTracker.cs

```diff
@@ -18,14 +18,15 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System;
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 
 namespace SonarAnalyzer.Helpers
 {
+    public delegate bool InvocationCondition(InvocationContext invocationContext);
```

Comment:
I know someone who will like this a lot ;)

Reply:
;) Delegates are a better choice here - _InvocationCondition_ more descriptive and shorter to type than _Func<InvocationContext, bool>_